hr_holidays_imposed_days/models/hr_holidays_imposed.py

```diff
@@ -0,0 +1,137 @@
+# Author: Damien Crier
+# Copyright 2016-2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import datetime
+import math
+
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import ValidationError
+
+HOURS_PER_DAY = 8
+
+
+class HrHolidaysImposed(models.Model):
+    _name = 'hr.holidays.imposed'
+    _description = 'Manage imposed holidays'
+
+    name = fields.Char(required=True)
+    date_from = fields.Datetime(required=True)
+    date_to = fields.Datetime(required=True)
+    number_of_days = fields.Float()
+    company_id = fields.Many2one(
+        'res.company',
+        string="Company",
+        default=lambda self: self.env['res.company']._company_default_get(
+            'hr.holidays.imposed')
+    )
+    status_id = fields.Many2one(
+        'hr.holidays.status',
+        string="Leave type",
+        required=True
+    )
+    employee_ids = fields.Many2many(
+        'hr.employee',
+        help="If empty, all employees of the "
+             "company will have this day/period "
+             "imposed")
+    auto_confirm = fields.Boolean()
+
+    @api.multi
+    def validate(self):
+        for rec in self:
+            created = self.env['hr.holidays']
+            if rec.employee_ids:
+                employees = rec.employee_ids
+            else:
+                employees = self.env['hr.employee'].search([
+                    ('company_id', '=', rec.company_id.id)
+                ])
+            for employee in employees:
+                vals = rec._prepare_leave_from_imposed_day()
+                vals.update({'employee_id': employee.id})
+                leave = self.env['hr.holidays'].create(vals)
```

Comment:
You may update it like:
line 42: holiday_obj = self.env['hr.holidays']
line 53: leave = holiday_obj.create(vals)